tests/components/heos/test_config_flow.py

```diff
@@ -47,36 +47,36 @@
 
 async def test_create_entry_when_host_valid(hass, controller):
     """Test result type is create entry when host is valid."""
-    flow = HeosFlowHandler()
-    flow.hass = hass
     data = {CONF_HOST: "127.0.0.1"}
-    result = await flow.async_step_user(data)
+    result = await hass.config_entries.flow.async_init(
+        heos.DOMAIN, context={"source": "user"}, data=data
+    )
     assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
     assert result["title"] == "Controller (127.0.0.1)"
     assert result["data"] == data
-    assert controller.connect.call_count == 1
+    assert controller.connect.call_count == 2
```

Comment:
We should patch `"homeassistant.components.heos.async_setup_entry"` so that the integration isn't set up here. We can assert that call to the mock from the patch.

The assertion here shouldn't change.